### PR TITLE
Add jenkins-operator for prow

### PR DESCRIPTION
**Pull request by lentzi90**
The name is a bit misleading unfortunately. This prow component is used to trigger jenkins jobs. It will allow us to have one common way to trigger all tests (except github workflows) and we can stop using the ghprb jenkins plugin that is no longer maintained.

### CODE REVIEW

File: prow/manifests/overlays/metal3/patches/jenkins-operator.yaml
Comment (lentzi90) [inline prow/manifests/overlays/metal3/patches/jenkins-operator.yaml:1]: This may be silly of me, but we have done all other resource requests as a separate patch so I did the same here also.

**Comment by lentzi90** [comment]
Not relevant
/override test-ubuntu-integration-main

**Comment by tuminoid** [comment]
Looks nice! How do we plan to test it?

**Comment by lentzi90** [comment]
The jenkins-operator has already been installed in the cluster, but the config change has not been applied. That will happen when we merge this.
To test these changes I have applied the jenkins-operator manifests manually and configured it to use a separate configmap instead of the auto-synced. Then I created a prow job and applied it to see that the jenkins-operator was able to trigger the jenkins job. This worked well.

The part that I could not test is the deck integration that should allow it to display logs from jenkins. However, I have confirmed that it is possible to get the logs by curling the jenkins-operator pod directly so I'm relatively sure that it will work.

After we merge this, we will still need to add trigger configurations for all jobs that we want to trigger in this way. I think we should do that separately though, and perhaps not all jobs at once. We could for example start with the BMO e2e jobs that are currently not required as a low risk first step.

**Comment by tuminoid** [comment]
Sounds fair! Timing just before holiday vacations though. It sounds like we can merge this, but maybe not mess with JJB and triggers just yet?

/approve

**Comment by lentzi90** [comment]
Yeah that makes sense!

**Review by smoshiur1237 — APPROVED** [review]
/lgtm
